redbot/cogs/mod/events.py

```diff
@@ -158,21 +159,31 @@ async def on_message(self, message):
         if not deleted:
             await self.check_mention_spam(message)
 
+    @staticmethod
+    def _update_past_names(name: str, name_list: List[Optional[str]]) -> None:
+        while None in name_list:  # clean out null entries from a bug
+            name_list.remove(None)
+        if name in name_list:
+            # Ensure order is maintained without duplicates occurring
+            name_list.remove(name)
+        name_list.append(name)
+        while len(name_list) > 20:
+            name_list.pop(0)
+
     @commands.Cog.listener()
     async def on_user_update(self, before: discord.User, after: discord.User):
         if before.name != after.name:
             track_all_names = await self.config.track_all_names()
             if not track_all_names:
                 return
             async with self.config.user(before).past_names() as name_list:
-                while None in name_list:  # clean out null entries from a bug
-                    name_list.remove(None)
-                if before.name in name_list:
-                    # Ensure order is maintained without duplicates occurring
-                    name_list.remove(before.name)
-                name_list.append(before.name)
-                while len(name_list) > 20:
-                    name_list.pop(0)
+                self._update_past_names(before.name, name_list)
+        if before.display_name != after.display_name:
+            track_all_names = await self.config.track_all_names()
+            if not track_all_names:
+                return
+            async with self.config.user(before).past_display_names() as name_list:
```

Comment:
```py
Traceback (most recent call last):
  File "venv\lib\site-packages\discord\client.py", line 441, in _run_event
    await coro(*args, **kwargs)
  File "venv\lib\site-packages\redbot\cogs\mod\events.py", line 185, in on_user_update
    async with self.config.user(before).past_display_names() as name_list:
  File "venv\lib\site-packages\redbot\core\config.py", line 363, in __getattr__
    raise AttributeError("'{}' is not a valid registered Group or value.".format(item))
AttributeError: 'past_display_names' is not a valid registered Group or value.
```
First thing in console when loading this on a bot. This key needs to be added to the defaults.